llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -17544,13 +17544,14 @@ AArch64TargetLowering::BuildSDIVPow2(SDNode *N, const APInt &Divisor,
                                      SmallVectorImpl<SDNode *> &Created) const {
   AttributeList Attr = DAG.getMachineFunction().getFunction().getAttributes();
   if (isIntDivCheap(N->getValueType(0), Attr))
-    return SDValue(N,0); // Lower SDIV as SDIV
+    return SDValue(N, 0); // Lower SDIV as SDIV
 
   EVT VT = N->getValueType(0);
 
   // For scalable and fixed types, mark them as cheap so we can handle it much
   // later. This allows us to handle larger than legal types.
-  if (VT.isScalableVector() || Subtarget->useSVEForFixedLengthVectors())
+  if (VT.isScalableVector() ||
+      (!VT.isScalarInteger() && Subtarget->useSVEForFixedLengthVectors()))
```

Comment:
So I had a fix for #97821 downstream as well and I believe here you actually want something like:

```
if (VT.isScalableVector() ||
    (VT.isFixedLengthVector() && Subtarget->useSVEForFixedLengthVectors())
```

since it's not obvious to me that `!VT.isScalarInteger()` is equivalent to `VT.isFixedLengthVector()`.

Reply:
Apply your comment, thanks